enhancements/agent-installer/automated-workflow-for-agent-based-installer.md

```diff
@@ -0,0 +1,616 @@
+service on node0.
+
+When all of the hosts are booted with the image, the client on node0
+will use the assisted service to trigger the deployment of the cluster
+(including applicable pre-flight validations) without further
+interaction from the user.
+
+The `agent create-image` command will produce as output credentials to
+access the cluster that is about to be created. This allows a client,
+such as a tool monitoring deployment progress, to use the cluster's
+API as soon as the API service is available.
+
+During deployment, the third-party orchestration tool will need a way
+to monitor progress and discover errors. The assisted service
+read-only API endpoints running on node0 will report that data. In the
+event of failure, a read-only endpoint in that service will be
+available to download any logs that have been collected.
+
+We will be taking advantage of the assisted service's ability to run
+the bootstrap services "in place" to avoid having a separate bootstrap
+VM. Bootstrap will run on node0, where the assisted service runs, to
+ensure that the service and its API are available for as long as
+possible.
+
+When bootstrapping completes and node0 reboots, progress monitoring
+will need to shift to the OpenShift APIs (`ClusterVersion` and
+`ClusterOperator`). We will extend the `openshift-install` binary with
+a new sub-command, `agent wait-for install-complete`, to use both APIs for monitoring
+and [progress
+reporting](https://github.com/openshift/assisted-service/blob/master/docs/enhancements/installation-progress-bar.md)
+in that case, including reporting when steps like validation fail and
+the cluster deployment is never going to succeed. *Users will still be
+able to write their own monitoring logic using the assisted service
+and OpenShift APIs directly, but they will have the option of using
+the command line tool.*
+
+For the fully automated workflow, the steps a user might normally
+perform such as approving agents for installation or triggering the
+installation will be handled by a new client running as a service on
+node0. The client will read the user's inputs from files included in
+the image and interact with the assisted-service REST API on localhost
+to drive the deployment. For the purposes of this document, we will
+refer to that service as a single entity called `create-cluster`,
+although in practice we may divide some of the steps into separate
+tools.
+
+### Actors in the Workflow
+
+**infrastructure admin** is a human user responsible for infrastructure
+within the data center, such as an image registry, networking, etc.
+
+**cluster creator** is a human user responsible for deploying a cluster.
+
+**orchestration tool** is intended to be a software component provided
+by the end user for managing the overall deployment process, including
+any steps for which the installer is not responsible. Those steps may
+also be performed by **cluster creator** if no software component is
+available.
+
```

Comment:
I am not sure what `Those steps may also be performed by **cluster creator** if no software component is available.` means. Do you mean the steps of deployment can be handled manually if no automation exists?